src/Microsoft.AspNet.Mvc.Extensions/Formatters/JsonOutputFormatter.cs

```diff
@@ -55,7 +65,12 @@ public void WriteObject([NotNull] TextWriter writer, object value)
             }
         }
 
-        private JsonWriter CreateJsonWriter(TextWriter writer)
+        /// <summary>
+        /// Called during serialization to get the <see cref="JsonWriter"/>.
+        /// </summary>
+        /// <param name="writer">The <see cref="TextWriter"/> used to write.</param>
+        /// <returns>The <see cref="JsonWriter"/> used during serialization.</returns>
+        public virtual JsonWriter CreateJsonWriter(TextWriter writer)
```

Comment:
@rynowak, I checked and looks like I've added these two methods as part of this https://github.com/aspnet/Mvc/commit/13e1c9d5e1509b0df5c7a2317b6380fa1a401784. And I can't find any reason why it has to be `public`. I'll change these to `protected`.